deploy/kubedirector/kubedirector.hpe.com_kubedirectorapps_crd.yaml

```diff
@@ -137,6 +137,14 @@ spec:
                     items:
                       type: string
                       pattern: '^configure$|^addnodes$|^delnodes$'
+                  containerSpec:
+                    type: object
+                    nullable: true
+                    properties:
+                      stdin:
+                        type: boolean
+                      tty:
+                        type: boolean
```

Comment:
So, if tty is True but stdin is False, the pod creation will fail. We handle that (as in things don't catch on fire or explode) but it would be nicer if admitClusterCR could catch that case up-front and reject such a spec.

Reply:
Sorry, I mean admitAppCR.

Reply:
Hmm, this is interesting. Did some experiments with tty true and stdin false. Looks like statefulset controller ignore tty in this case and goes ahead with pod creation, stateful set spec
```
spec:
  podManagementPolicy: OrderedReady
  replicas: 3
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: nginx
  serviceName: nginx
  template:
    metadata:
      creationTimestamp: null
      labels:
        app: nginx
    spec:
      containers:
      - image: k8s.gcr.io/nginx-slim:0.8
        imagePullPolicy: IfNotPresent
        name: nginx
        ports:
        - containerPort: 80
          name: web
          protocol: TCP
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        tty: true
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 10
  updateStrategy:
    rollingUpdate:
      partition: 0
    type: RollingUpdate
status:
  collisionCount: 0
  currentReplicas: 3
  currentRevision: web-6c7ff77887
  observedGeneration: 1
  readyReplicas: 3
  replicas: 3
  updateRevision: web-6c7ff77887
  updatedReplicas: 3
  
  ```

When I describe, I don't see tty in pod template 
```
Name:               web
Namespace:          hpecp
CreationTimestamp:  Thu, 15 Apr 2021 09:20:32 -0700
Selector:           app=nginx
Labels:             <none>
Annotations:        kubectl.kubernetes.io/last-applied-configuration:
                      {"apiVersion":"apps/v1","kind":"StatefulSet","metadata":{"annotations":{},"name":"web","namespace":"hpecp"},"spec":{"replicas":3,"selector...
Replicas:           3 desired | 3 total
Update Strategy:    RollingUpdate
  Partition:        824633989084
Pods Status:        3 Running / 0 Waiting / 0 Succeeded / 0 Failed
Pod Template:
  Labels:  app=nginx
  Containers:
   nginx:
    Image:        k8s.gcr.io/nginx-slim:0.8
    Port:         80/TCP
    Host Port:    0/TCP
    Environment:  <none>
    Mounts:       <none>
  Volumes:        <none>
Volume Claims:    <none>
Events:
  Type    Reason            Age    From                    Message
  ----    ------            ----   ----                    -------
  Normal  SuccessfulCreate  7m53s  statefulset-controller  create Pod web-0 in StatefulSet web successful
  Normal  SuccessfulCreate  7m51s  statefulset-controller  create Pod web-1 in StatefulSet web successful
  Normal  SuccessfulCreate  7m49s  statefulset-controller  create Pod web-2 in StatefulSet web successful

Reply:
To summarize, pod creation always succeeds irrespetive of tty/stdin combination. If this is the default behaviour should we still add validations at app launch time?

Reply:
Hmm. If they're not getting what they asked for then IMO yeah we should prevent it.

Reply:
ok sounds good.

Reply:
stdin without tty is ok? @joel-bluedata

Reply:
As far as I know. Just going from the K8s API spec.

Reply:
ok then, added validation in addAppCR